### PR TITLE
python312Packages.pixel-font-knife: init at 0.0.7

### PR DESCRIPTION
**Pull request by TakWolf**
## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by TakWolf** [comment]
https://github.com/NixOS/nixpkgs/pull/353750

File: pkgs/development/python-modules/pixel-font-knife/default.nix
Comment (FliegendeWurst) [inline pkgs/development/python-modules/pixel-font-knife/default.nix:1]: You need to add it to the python module list too

File: pkgs/development/python-modules/pixel-font-knife/default.nix [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+
+  nativeCheckInputs = [ pytestCheckHook ];
+
+  pythonImportsCheck = [ "pixel_font_knife" ];
+
+  passthru.updateScript = nix-update-script { };
+
+  meta = {
+    homepage = "https://github.com/TakWolf/pixel-font-knife";
+    description = "A set of pixel font utilities";
```
Comment (FliegendeWurst) [inline pkgs/development/python-modules/pixel-font-knife/default.nix:39]:
```suggestion
    description = "Set of pixel font utilities";
```

**Review by FliegendeWurst — CHANGES_REQUESTED** [review]

File: pkgs/development/python-modules/pixel-font-knife/default.nix [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+    pname = "pixel_font_knife";
+    inherit version;
+    hash = "sha256-NALIf8DbIPKmazF669LHsKuP0tndR0pbphrhlXt6wyA=";
+  };
+
+  build-system = [ hatchling ];
+
+  dependencies = [
+    pypng
+  ];
```
Comment (FliegendeWurst) [inline pkgs/development/python-modules/pixel-font-knife/default.nix:29]: Build fails with
```
       > Executing pythonRuntimeDepsCheck
       > Checking runtime dependencies for pixel_font_knife-0.0.7-py3-none-any.whl
       >   - pyyaml not installed
       >   - unidata-blocks not installed
```

File: pkgs/development/python-modules/pixel-font-knife/default.nix [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+    pypng
+    unidata-blocks
+    pyyaml
+  ];
+
+  nativeCheckInputs = [ pytestCheckHook ];
+
+  pythonImportsCheck = [ "pixel_font_knife" ];
+
+  passthru.updateScript = nix-update-script { };
```
Comment (FliegendeWurst) [inline pkgs/development/python-modules/pixel-font-knife/default.nix:39]: I think all Python modules already have an update script (based on pypi I think)

**Review by FliegendeWurst — APPROVED** [review]

File: pkgs/development/python-modules/pixel-font-knife/default.nix (hunk 3 above)
Comment (TakWolf) [reply pkgs/development/python-modules/pixel-font-knife/default.nix:39]: I am not a direct user of nix, I am referring to the existing configuration.

https://github.com/NixOS/nixpkgs/blob/master/pkgs/development/python-modules/bdffont/default.nix#L30

I'm not sure how to modify it, can you modify it directly

Comment (FliegendeWurst) [inline pkgs/development/python-modules/pixel-font-knife/default.nix:39]:
```suggestion
```

Comment (TakWolf) [reply pkgs/development/python-modules/pixel-font-knife/default.nix:39]: Let me confirm, the pypi module can automatically update without this line, right?

Comment (FliegendeWurst) [reply pkgs/development/python-modules/pixel-font-knife/default.nix:39]: Yes

**Review by FliegendeWurst — APPROVED** [review]

**Pull request merged**